### PR TITLE
JavaScript_week-2_Erhan-Koral

### PR DESCRIPTION
**Pull request by ErhanKRL**

### CODE REVIEW

File: 1-JavaScript/Week2/assignment/ex1-giveCompliment.js [hunk 1]
```diff
@@ -17,14 +17,30 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/1-Java
    your name.
    Use `console.log` each time to display the return value of the
    `giveCompliment` function to the console.
+
 -----------------------------------------------------------------------------*/
-function giveCompliment(/* TODO parameter(s) go here */) {
-  // TODO complete this function
```
Comment (gabsalvo) [inline 1-JavaScript/Week2/assignment/ex1-giveCompliment.js:22]: Great!

File: 1-JavaScript/Week2/assignment/ex2-dogYears.js [hunk 2]
```diff
@@ -16,8 +16,9 @@ calculate it!
    ages.
 -----------------------------------------------------------------------------*/
 
-function calculateDogAge(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function calculateDogAge(number) {
```
Comment (gabsalvo) [inline 1-JavaScript/Week2/assignment/ex2-dogYears.js:19]: Keep it up!!

File: 1-JavaScript/Week2/assignment/ex3-tellFortune.js [hunk 3]
```diff
@@ -33,30 +33,26 @@ body, this code is now written once only in a separated function.
...
 // a randomly selected element as its return value.
-function selectRandomly(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function selectRandomly(array) {
+  return array[Math.floor(Math.random() * array.length)];
 }
 
-function tellFortune(/* TODO add parameter(s) here */) {
-  // TODO complete this function
+function tellFortune(numKids, partnerName, location, jobTitle) {
```
Comment (gabsalvo) [inline 1-JavaScript/Week2/assignment/ex3-tellFortune.js:40]: Good good! Try to make it even more modular by adding some const for each function call you do! But great work

File: 1-JavaScript/Week2/assignment/ex4-shoppingCart.js [hunk 4]
```diff
@@ -23,8 +23,16 @@ you have more than 3 items in your shopping cart the first item gets taken out.
 const shoppingCart = ['bananas', 'milk'];
 
 // ! Function to be tested
-function addToShoppingCart(/* parameters go here */) {
-  // TODO complete this function
+function addToShoppingCart(item) {
+  if (item) {
+    if (shoppingCart.length >= 3) {
```
Comment (gabsalvo) [inline 1-JavaScript/Week2/assignment/ex4-shoppingCart.js:28]: That's almost correct. You need to invert the order, you need to push the item before checking if it is length > 3

File: 1-JavaScript/Week2/assignment/ex5-shoppingCartPure.js [hunk 5]
```diff
@@ -16,8 +16,18 @@ it pure. Do the following:
 5. Confirm that you function passes the provided unit tests.
 ------------------------------------------------------------------------------*/
 // ! Function under test
-function addToShoppingCart(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function addToShoppingCart(shoppingCart, item) {
```
Comment (gabsalvo) [inline 1-JavaScript/Week2/assignment/ex5-shoppingCartPure.js:19]: That's correct but for the sake of being a real pro at js, try to improve it just by using the spread operator and a library function on arrays. Any ideas?

File: 1-JavaScript/Week2/assignment/ex6-totalCost.js [hunk 6]
```diff
@@ -21,22 +21,30 @@ instead!
...
+  chips: 0.99,
+  beer: 1.75,
+  cake: 5.56,
+  wine: 16.1,
+  peanut: 4.39,
 };
 
-function calculateTotalPrice(/* TODO parameter(s) go here */) {
-  // TODO replace this comment with your code
+function calculateTotalPrice(obj) {
```
Comment (gabsalvo) [inline 1-JavaScript/Week2/assignment/ex6-totalCost.js:31]: I like this more than the provided solution ahahhaha

File: 1-JavaScript/Week2/assignment/ex7-mindPrivacy.js [hunk 7]
```diff
@@ -30,8 +30,12 @@ const employeeRecords = [
 ];
 
 // ! Function under test
-function filterPrivateData(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function filterPrivateData(employeeRecord) {
```
Comment (gabsalvo) [inline 1-JavaScript/Week2/assignment/ex7-mindPrivacy.js:33]: NICE!!!!!!!!!!!

File: 1-JavaScript/Week2/assignment/ex5-shoppingCartPure.js (hunk 5 above)
Comment (ErhanKRL) [reply 1-JavaScript/Week2/assignment/ex5-shoppingCartPure.js:19]: Yep. Ternary operator I guess.

Comment (gabsalvo) [reply 1-JavaScript/Week2/assignment/ex5-shoppingCartPure.js:19]: you could use slice
so you would have return [...shoppingCart,item].slice(-3)

**Pull request merged**